device_backends/LogicalNameMapping/src/LNMDoubleBufferPlugin.cc

```diff
@@ -130,7 +130,10 @@ namespace ChimeraTK { namespace LNMBackend {
 
   template<typename UserType>
   void DoubleBufferAccessorDecorator<UserType>::doPreRead(TransferType type) {
-    (*_plugin._readerCount)++;
+    {
+      std::lock_guard lg{_plugin._readerCount->mutex};
+      _plugin._readerCount->value++;
+    }
     // acquire a lock in firmware (disable buffer swapping)
     _enableDoubleBufferReg->accessData(0) = 0;
```

Comment:
I would only access this register while holding the mutex. At this place it is not strictly necessary because you are only conflicting with other threads which are writing the same value. But for simplicity I would formally protect this register with the mutex.